### PR TITLE
graphML Syntactic Checker: General graphML Syntax Error

### PR DESCRIPTION
**Pull request by ebachoura**
#363 As discussed on this issue, before this pull request, out servers crash whenever we input a graphML file with any sort of major syntactic error that is caught by the parser that we are using. I have created an error called the General GraphML Syntax Error and made it so that the very first thing that we check for on import is to make sure that there are no errors spewed out by the parser, and if so, then instantly return a network with a General GraphML Syntax Error. Let it be noted that not every syntactic error is caught by this. There is a full list of errors that we need to check for individually and some that are not caught by the parser at all. Find this list under #363

### CODE REVIEW

File: server/controllers/importers/graphml.js [hunk 1]
```diff
@@ -1,19 +1,84 @@
...
 var semanticChecker = require(__dirname + "/../semantic-checker");
 
+var graphmlWarnings = {
+  EDGES_WITHOUT_WEIGHTS: {
+    warningCode: "EDGES_WITHOUT_WEIGHTS",
+    errorDescription: "GRNsight has detected that one or more edges in your network are missing numerical weight" +
+    " values. Because the algorithm GRNsight uses for determining the arrowhead type and the color and thickness of" +
+    " the edges requires numerical weight values, your graph will display as an unweighted graph with black edges" +
+    " and pointed arrowheads. If you want to display the network as a weighted graph, please modify your input file" +
+    " to include weight values for all edges."
```
Comment (dondi) [inline server/controllers/importers/graphml.js]: Indent unfinished expressions, to show that those lines are “under” the `errorDescription` property name.

File: server/controllers/importers/graphml.js [hunk 2]
```diff
@@ -1,19 +1,84 @@
...
-    graph = result.graphml && result.graphml.graph && result.graphml.graph[0];
+    if (err) {
+      pushRelevantError(err);
+    } else {
+      key = result.graphml && result.graphml.key;
+      graph = result.graphml && result.graphml.graph && result.graphml.graph[0];
+    }
   });
 
+  if (network.errors.length > 0) { return semanticChecker(network); }
```
Comment (dondi) [inline server/controllers/importers/graphml.js]: Our style dictates that we break this into multiple lines:

    if (network.errors.length > 0) {
        return semanticChecker(network);
    }

File: server/controllers/importers/graphml.js [hunk 3]
```diff
@@ -1,19 +1,84 @@
...
+  function pushRelevantError(err) {
+    var parseError = readErrorFromErr(err);
+    network.errors.push(graphmlErrors.GRAPHML_GENERAL_SYNTAX_ERROR(parseError));
+    // switch(parseError) {
+    //   case "Invalid attribute name":
+    //     network.errors.push(graphmlErrors.UNKNOWN_ERROR);
+    //     break;
+    //   default:
+    //     network.errors.push(graphmlErrors.GRAPHML_GENERAL_SYNTAX_ERROR);
+    // }
```
Comment (dondi) [inline server/controllers/importers/graphml.js]: Add a comment here for next semester:

    // TODO Ask Dondi about how to do this in a more data-driven manner.

**Comment by ebachoura** [comment]
@dondi Alright I made the changes, let me know if there is anything else.

**Comment by kdahlquist** [comment]
As noted on issue #428, an empty .graphml file crashes the server.  Because @ebachoura is not around this summer, I was wondering if there is a quick fix to adding a test case for this instance since we know that it has a bad outcome.

**Pull request merged**